File: 백준/Bronze/16396. 선 그리기/선 그리기.py
import sys

input = sys.stdin.readline

result = 0
check = list(False for _ in range(10000))

N = int(input().strip())
for _ in range(N):
    X, Y = map(int, input().strip().split())
    for i in range(X, Y):
        check[i] = True

for ch in check:
    if ch:
        result += 1

sys.stdout.write(str(result) + '\n')

File: 백준/Bronze/16396. 선 그리기/선 그리기.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int result = 0;
        boolean[] check = new boolean[10000];

        int N = Integer.parseInt(br.readLine().trim());
        for(int i = 0; i < N; i++){
            String[] input = br.readLine().trim().split(" ");
            int X = Integer.parseInt(input[0]);
            int Y = Integer.parseInt(input[1]);
            for(int j = X; j < Y; j++){
                check[j] = true;
            }
        }

        for(boolean ch : check){
            if(ch) result++;
        }

        bw.write(String.valueOf(result));
        bw.flush();
        br.close();
        bw.close();
    }
}